src/app/portlets/dot-templates/dot-template-list/dot-template-list.component.ts

```diff
@@ -86,10 +147,351 @@ export class DotTemplateListComponent implements OnInit, OnDestroy {
+                    }
+                ]
+            }
+        };
+    }
+
+    private setTemplateBulkActions(): MenuItem[] {
+        return [
+            {
+                label: this.dotMessageService.get('Publish'),
+                command: () => {
+                    this.publishTemplate(
+                        this.selectedTemplates.map((template) => template.identifier)
+                    );
+                }
+            },
+            ...this.setLicenseAndRemotePublishTemplateBulkOptions(),
+            {
+                label: this.dotMessageService.get('Unpublish'),
+                command: () => {
+                    this.unPublishTemplate(
+                        this.selectedTemplates.map((template) => template.identifier)
+                    );
+                }
+            },
+            {
+                label: this.dotMessageService.get('Archive'),
+                command: () => {
+                    this.archiveTemplates(
+                        this.selectedTemplates.map((template) => template.identifier)
+                    );
+                }
+            },
+            {
+                label: this.dotMessageService.get('Unarchive'),
+                command: () => {
+                    this.unArchiveTemplate(
+                        this.selectedTemplates.map((template) => template.identifier)
+                    );
+                }
+            },
+            {
+                label: this.dotMessageService.get('Delete'),
+                command: () => {
+                    this.deleteTemplate(
+                        this.selectedTemplates.map((template) => template.identifier)
+                    );
+                }
             }
         ];
     }
+
+    private setCopyTemplateOptions(template: DotTemplate): DotActionMenuItem[] {
+        return template.canWrite
+            ? [
+                  {
+                      menuItem: {
+                          label: this.dotMessageService.get('Copy'),
+                          command: () => {
+                              this.dotTemplatesService
+                                  .copy(template.identifier)
+                                  .pipe(take(1))
+                                  .subscribe((response: DotTemplate) => {
+                                      if (response) {
+                                          this.showToastNotification(
+                                              this.dotMessageService.get('message.template.copy')
+                                          );
+                                          this.listing.loadCurrentPage();
+                                      }
+                                  });
+                          }
+                      }
+                  }
+              ]
+            : [];
+    }
+
+    private setUnlockTemplateOptions(template: DotTemplate): DotActionMenuItem[] {
+        return template.locked && template.canWrite
```

Comment:
Shouldn't we prevent this call under the condition?

Reply:
only if both are true we add the option.